src/plots/plots.js

```diff
@@ -1866,8 +1866,8 @@ function initMargins(fullLayout) {
     if(!fullLayout._pushmarginIds) fullLayout._pushmarginIds = {};
 }
 
-var minFinalWidth = 64; // could possibly be exposed as layout.margin.minfinalwidth
-var minFinalHeight = 64; // could possibly be exposed as layout.margin.minfinalheight
```

Comment:
I think we should define two values for each dimension here (so repeat this for width as well):
```js
// non-negotiable - this is the smallest height we will allow users to specify via explicit margins
var MIN_SPECIFIED_HEIGHT = 2;

// could be exposed as an option - the smallest we will allow automargin to shrink a larger plot
var MIN_REDUCED_HEIGHT = 64;
```
Then down below we can combine these two:
```js
var minFinalHeight = Lib.constrain(
    height - margin.t - margin.b,
    MIN_SPECIFIED_HEIGHT,
    MIN_REDUCED_HEIGHT
);
```
The goal is:
- if you specify a plot height larger than `MIN_REDUCED_HEIGHT`, automargin is allowed to _increase_ the top and bottom margins to reduce the plot height as low as `MIN_REDUCED_HEIGHT`.
- if you specify a plot area smaller than `MIN_REDUCED_HEIGHT` but larger than `MIN_SPECIFIED_HEIGHT`, automargin will make no changes.
- if you specify a plot height smaller than `MIN_SPECIFIED_HEIGHT`, automargin should _reduce_ the margins symmetrically to increase the plot height back to `MIN_SPECIFIED_HEIGHT`